### PR TITLE
clientv3: specify watch progress notify interval

### PR DESCRIPTION
**Pull request by gyuho**
For watch request

/cc @heyitsanthony @xiang90 

Otherwise, people have to search through our code base to find out the interval.

### CODE REVIEW

**Comment by gyuho** [comment]
Btw is this working?

```go
package main

import (
	"fmt"
	"log"
	"time"

	"github.com/coreos/etcd/clientv3"
	"github.com/coreos/etcd/etcdserver/api/v3rpc"
	"golang.org/x/net/context"
)

var (
	dialTimeout    = 5 * time.Second
	requestTimeout = 1 * time.Second
	endpoints = []string{"localhost:2379"}
)

func main() {
	cli, err := clientv3.New(clientv3.Config{
		Endpoints:   endpoints,
		DialTimeout: dialTimeout,
	})
	if err != nil {
		log.Fatal(err)
	}
	defer cli.Close()

	wc := clientv3.NewWatcher(cli)
	defer wc.Close()

	rch := wc.Watch(context.Background(), "foo", clientv3.WithProgressNotify())
	v3rpc.SetProgressReportInterval(2 * time.Second)
	
	i := 0
	for i != 30 {
		fmt.Println()
		fmt.Println("i:", i)
		select {
		case wresp := <-rch:
			fmt.Printf("%+v\n\n", wresp)
			for _, ev := range wresp.Events {
				fmt.Printf("%s %q : %q (mod rev %d, create rev %d)\n", ev.Type, ev.Kv.Key, ev.Kv.Value, ev.Kv.ModRevision, ev.Kv.CreateRevision)
			}
		}
		time.Sleep(time.Second)
		i++
	}
}
```

I don't get updates every seconds?

**Comment by xiang90** [comment]
This will not work. The notification interval is fixed.

**Comment by gyuho** [comment]
@xiang90 Even with `v3rpc.SetProgressReportInterval(2 * time.Second)`?

**Comment by xiang90** [comment]
@gyuho No. You you are setting a server side configuration, which is designed for testing.

**Comment by gyuho** [comment]
@xiang90 Ah, right. Just fixed the comment, without the set interval. PTAL. Thanks.

File: clientv3/op.go [hunk 1]
```diff
@@ -289,6 +289,7 @@ func withTop(target SortTarget, order SortOrder) []OpOption {
 
 // WithProgressNotify makes watch server send periodic progress updates.
```
Comment (xiang90) [inline clientv3/op.go]: send periodic progress updates every 10 minutes even if there is no actual events.

File: clientv3/op.go [hunk 2]
```diff
@@ -287,7 +287,8 @@ func withTop(target SortTarget, order SortOrder) []OpOption {
 	return []OpOption{WithPrefix(), WithSort(target, order), WithLimit(1)}
 }
 
-// WithProgressNotify makes watch server send periodic progress updates.
+// WithProgressNotify makes watch server send periodic progress updates
+// every 10 minutes even if there is no actual events.
```
Comment (xiang90) [inline clientv3/op.go]: when there is no incoming events?

**Comment by gyuho** [comment]
@xiang90 PTAL. Thanks!

**Comment by xiang90** [comment]
lgtm

**Pull request merged**